src/utils/index.ts

```diff
@@ -2,3 +2,26 @@ export function getUrlOriginWithPath(url: string) {
     const { origin, pathname } = new URL(url);
     return new URL(pathname, origin).toString();
 }
+
+export function toError(value: unknown): Error {
+    if (value instanceof Error) {
+        return value;
+    }
+
+    if (typeof value === 'undefined') {
+        return new Error();
+    }
+
+    let errorMessage = String(value);
```

Comment:
it is weird to unconditionally convert potential object into string, if we can avoid it

```
export function toError(value: unknown): Error {
    if (value instanceof Error) {
        return value;
    }

    if (typeof value === 'undefined') {
        return new Error();
    }

    if (typeof value === 'object' && value !== null) {
        if ('message' in value) return new Error(String(value.message));

        if ('data' in value) {
            return new Error(String(value.data));
        }
    }

    return new Error(String(value));
}
```

Reply:
@olehrakwix please review slightly refactored `toError` util